tools/pubsys-config/src/lib.rs

```diff
@@ -49,10 +49,31 @@ impl InfraConfig {
             Ok(Self::default())
         }
     }
+
+    /// Deserializes an InfraConfig.lock to yaml format from a given path
+    pub fn from_path_yaml<P>(path: P) -> Result<Self>
```

Comment:
Should just be `Infra.lock` and not `InfraConfig.lock` I think?

Also do we really need this to be a method? We're only calling this in a unit test right now. 
For `CheckInfraConfig` subcommand we would need it I guess.

Also `yaml` is mostly an implementation detail. We're just deserializing the lock file from path.
```suggestion
    /// Deserializes an InfraConfig from a Infra.lock file at a given path
    pub fn from_lock_path<P>(path: P) -> Result<Self>
```